src/plugins/data_source_management/public/constants.ts

```diff
@@ -29,3 +31,30 @@ export const EDIT = '/edit';
 export const DATACONNECTIONS_UPDATE_STATUS = '/status';
 export const INTEGRATIONS_BASE = '/api/integrations';
 export const observabilityMetricsID = 'observability-metrics';
+
+// Module for handling EMR states for Dashboards Progress Bar. All of these except "initial" are
+// directly from the EMR job run states. "ord" is used to approximate progress (eyeballed relative
+// stage times), and "terminal" indicates whether a job is in progress at all.
+export const EMR_STATES = new Map<DirectQueryLoadingStatus, { ord: number; terminal: boolean }>([
+  [DirectQueryLoadingStatus.SUBMITTED, { ord: 0, terminal: false }],
+  [DirectQueryLoadingStatus.SCHEDULED, { ord: 30, terminal: false }],
+  [DirectQueryLoadingStatus.RUNNING, { ord: 70, terminal: false }],
+  [DirectQueryLoadingStatus.SUCCESS, { ord: 100, terminal: true }],
+  [DirectQueryLoadingStatus.FAILED, { ord: 100, terminal: true }],
+  [DirectQueryLoadingStatus.CANCELED, { ord: 100, terminal: true }],
+  [DirectQueryLoadingStatus.INITIAL, { ord: 100, terminal: true }],
+]);
+
+export const MAX_ORD = 100;
+
+export function intervalAsMinutes(interval: number): string {
```

Comment:
how is this used? if this is for non client facing

would it be more precise to go with nano or ms?